flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/factories/DataGenTableSourceFactoryOptions.java

```diff
@@ -0,0 +1,97 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.factories;
+
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/** {@link ConfigOption}s for {@link DataGenTableSourceFactory}. */
+public class DataGenTableSourceFactoryOptions {
+
+    public static final Long ROWS_PER_SECOND_DEFAULT_VALUE = 10000L;
+
+    public static final String FIELDS = "fields";
+    public static final String KIND = "kind";
+    public static final String START = "start";
+    public static final String END = "end";
+    public static final String MIN = "min";
+    public static final String MAX = "max";
+    public static final String LENGTH = "length";
+
+    public static final String SEQUENCE = "sequence";
+    public static final String RANDOM = "random";
+
+    public static final ConfigOption<Long> ROWS_PER_SECOND =
+            key("rows-per-second")
+                    .longType()
+                    .defaultValue(ROWS_PER_SECOND_DEFAULT_VALUE)
+                    .withDescription("Rows per second to control the emit rate.");
+
+    public static final ConfigOption<Long> NUMBER_OF_ROWS =
+            key("number-of-rows")
+                    .longType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Total number of rows to emit. By default, the source is unbounded.");
+
+    /** Placeholder {@link ConfigOption}. Not used for retrieving values. */
+    public static final ConfigOption<String> FIELD_KIND =
+            ConfigOptions.key(String.format("%s.#.%s", FIELDS, KIND))
+                    .stringType()
+                    .defaultValue("random")
+                    .withDescription("Generator of this '#' field. Can be 'sequence' or 'random'.");
+
+    /** Placeholder {@link ConfigOption}. Not used for retrieving values. */
+    public static final ConfigOption<String> FIELD_MIN =
+            ConfigOptions.key(String.format("%s.#.%s", FIELDS, MIN))
+                    .stringType()
+                    .defaultValue("(Minimum value possible for the type of the field)")
+                    .withDescription("Minimum value to generate for fields of kind 'random'.");
+
+    /** Placeholder {@link ConfigOption}. Not used for retrieving values. */
+    public static final ConfigOption<String> FIELD_MAX =
+            ConfigOptions.key(String.format("%s.#.%s", FIELDS, MAX))
+                    .stringType()
+                    .defaultValue("(Maximum value possible for the type of the field)")
```

Comment:
nit: shall we do `.noDefaultValue()` for this and the option above? It could explode the website formatting and is actually obvious or could be moved to the description.